Expensify/App
# Store User_IsFromPublicDomain in Onyx

### Details
This is a weird API that relies on Bedrock cache. If there is no domain info in the bedrock cache, we create a separate bedrock job to get the domain info from clearbit. In this case, the `User.getPublicDomainInfo` Onyx action will auto-retry itself after 10 minutes.

### Fixed Issues
Fixes (partial) https://github.com/Expensify/Expensify/issues/120480

### Tests / QA Steps (Web/Desktop)
1. Login to Expensify.cash as a public domain user, such as `roryabrahamtest@gmail.com`
1. Give the app a few seconds to load, then open the JS console, and look in `Application` -> `Local Storage`. Look for a key "user", and verify that it contains the data `isFromPublicDomain: true`.
1. Login to Expensify.cash as a private domain user, such as `rory@expensify.com`
1. Give the app a few seconds to load, then open the JS console, and look in `Application` -> `Local Storage`. Look for a key "user", and verify that it contains the data `isFromPublicDomain: false`.

#### On dev only
1. Verify that your dev console shows debug-level logs
1. Create and validate a new account with a private domain name, such as `rory@mynewdomain.com`
1. Login to E.cash with the new account.
1. Check the local storage – there should be no value `isFromPublicDomain` under the `user` key.
1. There should also be a debug-level console log saying:

    ```
    Command User_IsFromPublicDomain returned error code: 666. Most likely, this means that the domain mynewdomain.com is not in the bedrock cache. Retrying in 10 minutes
    ```
    
1. Go into the bedrock sql database, and check for a bedrock job:

    ```sql
    SELECT * FROM jobs WHERE name = 'www-prod/ClearbitCheckPublicEmail?domain=mynewdomain.com'
    ```
    
1. Run `bwm.sh`
1. After 10 minutes has passed, check the local storage again. Verify that the value `isFromPublicDomain: false` is set under the `user` key.

### Tests / QA (iOS/Android/mWeb)
QA is dependent on easy access to JS console, so just regular regression testing - no extra QA.

### Tested On


### Screenshots

#### Web
![image](https://user-images.githubusercontent.com/47436092/121399571-674f3700-c90b-11eb-884c-b2f12f331348.png)

### Desktop
![image](https://user-images.githubusercontent.com/47436092/121410034-02014300-c917-11eb-8020-072feed36b2a.png)
